repos/system_upgrade/el7toel8/actors/selinux/selinuxcontentscanner/actor.py

```diff
@@ -0,0 +1,193 @@
+# to be used with grep
+REMOVED_TYPES="|".join(REMOVED_TYPES_)
+# to be used with sed
+SED_COMMAND='/' + '\|'.join(REMOVED_TYPES_) + '/s/^/;/g'
+
+# types, attributes and boolean contained in container-selinux
+CONTAINER_TYPES="|".join(["container_connect_any","container_runtime_t","container_runtime_exec_t","spc_t","container_auth_t","container_auth_exec_t","spc_var_run_t","container_var_lib_t","container_home_t","container_config_t","container_lock_t","container_log_t","container_runtime_tmp_t","container_runtime_tmpfs_t","container_var_run_t","container_plugin_var_run_t","container_unit_file_t","container_devpts_t","container_share_t","container_port_t","container_build_t","container_logreader_t","docker_log_t","docker_tmpfs_t","docker_share_t","docker_t","docker_lock_t","docker_home_t","docker_exec_t","docker_unit_file_t","docker_devpts_t","docker_config_t","docker_tmp_t","docker_auth_exec_t","docker_plugin_var_run_t","docker_port_t","docker_auth_t","docker_var_run_t","docker_var_lib_t","container_domain","container_net_domain"])
+
+
+class SELinuxContentScanner(Actor):
+    '''
+    Scan the system for any SELinux customizations
+
+    Find SELinux policy customizations (custom policy modules and changes
+    introduced by semanage) and save them in SELinuxModules and SELinuxCustom
+    models. Customizations that are incompatible with SELinux policy on RHEL-8
+    are removed.
+    '''
+    name = 'selinuxcontentscanner'
+    consumes = (SELinuxFacts, )
+    produces = (SELinuxModules, SELinuxCustom, )
+    tags = (FactsPhaseTag, IPUWorkflowTag, )
+
+    def process(self):
+        # exit if SELinux is disabled
+        for fact in self.consume(SELinuxFacts):
+            if not fact.enabled:
+                return
+
+        semodule_list = self.getSELinuxModules()
+
+        self.produce(SELinuxModules(modules=semodule_list))
+
+        semanage_removed = []
+        semanage_valid = []
+        try:
+            # Collect SELinux customizations and select the ones that
+            # can be reapplied after the upgrade
+            semanage = call(['semanage', 'export'], split=True)
+            for line in semanage:
+                for setype in REMOVED_TYPES_:
+                    if setype in line:
+                        semanage_removed.append(line)
+                        break
+                else:
+                    semanage_valid.append(line)
+
+        except subprocess.CalledProcessError:
+            return
+
+        self.produce(
+            SELinuxCustom(
+                commands=semanage_valid,
+                removed=semanage_removed
+            )
+        )
+
+    def checkModule(self, name):
+        """ Check if given module contains one of removed types.
+            If so, comment out corresponding lines and return them.
+            The function expects a text file "$name" containing cil policy
+            to be present in the current directory.
```

Comment:
Consider following of [PEP 257](https://www.python.org/dev/peps/pep-0257) with exceptions mentioned in contribution guidelines:
- https://github.com/oamg/leapp-repository/blob/master/CONTRIBUTING.md#code-style

It would help to the others and code will be more unified with the rest of the project. Similar for other descriptions in the code.